src/views/Search.vue

```diff
@@ -152,6 +152,9 @@ const filteredItems = function (itemType: string) {
 };
 
 onMounted(() => {
+  eventbus.on('globalsearchterm', (evt: GlobalSearchTermEvent) => {
```

Comment:
You mean you want a route to handle /search?searchterm="blah"?

Can you router.push to this route twice? Thinking of the scenario if you click stream title to search, then mistakenly press the mini player to go full screen, then press the stream title again to search - will this route?

Reply:
yeah, I'm not sure about that either tbh :-)
I'll have to do some tests. Its kind of cool that you can have the searchterm in the url as path query param

Reply:
Don't waste any time on this -  I'll have a play around with the router see what I can come up with and get back to you if I run into a wall.

Reply:
I've gave this a bit more thought and I can't really think of any benefits of using a path query param in a SPA.

Currently performing a manual search does not alter the URL to contain the search term.
If this is implemented for searching stream_title, the URL will have the original search term, but performing further searches will not update the URL, which is a bit confusing for the user.
On my phone, in Chrome I've done the "Add to Homescreen" option which hides the address bar completely and makes it behave like an app, which works very well indeed and gives a nice experience - so the search URL would not even be seen.

So I'm thinking using the eventbus is the cleaner and less confusing option for the user - what do you think?

Reply:
Yeah in that case it doesn't make any sense but still the eventbus is strange to communicate parameters to a page. I would just use the properties for that. You can call the router and tell the properties to send. (ab)using the eventbus or a shared stored is more for dialogs.

This is for example already use din ItemsListing.vue where we redirect an in-library search to the global search.

![Scherm­afbeelding 2024-04-27 om 11 37 36](https://github.com/music-assistant/frontend/assets/6389780/14adcad1-c371-43e0-a7c0-ce5180109a30)

Reply:
Yes I tried this, but remember you can't router.push to the existing route.  If /search is already the active route behind the full screen player, pressing the stream title to search for it will not work with router.push again.  In this case we need a way to communicate between two active components which have no direct ancestry.

https://vuejs.org/guide/components/events.html mentions resorting to an external event bus is sometimes needed:

>Unlike native DOM events, component emitted events do not bubble. You can only listen to the events emitted by a direct child component. If there is a need to communicate between sibling or deeply nested components, use an external event bus or a [global state management solution](https://vuejs.org/guide/scaling-up/state-management).

Reply:
hmmm, I still don't like it that as now there are multiple ways to do the same.

So we should either find a way to accept a router push/replace with updated params or switch to eventbus from the other components as well.

I have an idea... what about we replace the private searchterm in the search component with a global one on the store ? That way you can change the global property from anywhere and contain all logic inside search. Just add a watcher in there to do the actual search and store it to the localstorage etc.

Reply:
Update: I have updated the current implementation to the global variable in the store so if you rebase this branch you should be able to build on top of that.

Reply:
Ah ok so this is using the other suggestion in the vue docs for communicating between components by using a global state object any component can access.  Good idea, this should make it nice and consistent then!

I've rebased and updated to use the new solution.